modules/gdscript/gdscript_editor.cpp

```diff
@@ -1813,7 +1813,7 @@ static bool _guess_expression_type(GDScriptParser::CompletionContext &p_context,
 						}
 					}
 
-					if (!found && base.value.get_type() != Variant::NIL) {
+					if (!found && (base.value.get_type() != Variant::NIL || base.type.is_set())) {
```

Comment:
Forgive me for being thorough and asking potentially silly questions.

Wouldn't `base.type.is_set()` be enough of a check here? And why would it not be?

Reply:
Good point. I'd also guess `base.type.is_set` would be enough. Will test it.

Reply:
Ok it seems to work well without `base.value.get_type() != Variant::NIL`. However I'm unsure whether we should delete the original condition.

Hypothetical situation which might make problems (unsure whether it can be constructed):
- base type is `UNRESOLVED`
- but there is a value associated with the base

I don't now whether this is a possible scenario. Some insight from the GDScript team might help.

Reply:
I just removed the check completely. `_guess_method_return_type_from_base` does not use base.value and if `base.type.is_set() == false` `_guess_method_return_type_from_base` returns false without much performance overhead anyways.

This seems like the solution that is least likely to cause regressions if the implementation of `_guess_method_return_type_from_base` changes.